### PR TITLE
Fix inconsistent comment message

### PR DESCRIPTION
**Pull request by m1093782566**
**What this PR does / why we need it**:

remove dead code in `/pkg/proxy/apis/kubeproxyconfig/validation/validation.go` - see duplicated codes in https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/apis/kubeproxyconfig/types.go#L155-L169

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/proxy/apis/kubeproxyconfig/v1alpha1/types.go [hunk 1]
```diff
@@ -148,14 +148,11 @@ type KubeProxyConfiguration struct {
...
+// (newer, faster), 'ipvs', and 'kernelspace' (Windows only, newer).
+//
+// If blank, use the best-available proxy (currently iptables, but may change in
+// future versions). If the iptables proxy is selected, regardless of how, but
+// the system's kernel or iptables versions are insufficient, this always falls
+// back to the userspace proxy.
 type ProxyMode string
-
-const (
-	ProxyModeUserspace ProxyMode = "userspace"
```
Comment (bowei) [inline pkg/proxy/apis/kubeproxyconfig/v1alpha1/types.go]: Why not add the four types and reference them instead?

Comment (m1093782566) [reply pkg/proxy/apis/kubeproxyconfig/v1alpha1/types.go]: It's in `v1alpha1` package. IIUC, we seldom reference versioned const in validation.

Comment (m1093782566) [reply pkg/proxy/apis/kubeproxyconfig/v1alpha1/types.go]: BTW, I find there is no default value for `ProxyMode` in `pkg/proxy/apis/kubeproxyconfig/v1alpha1/defaults.go`, we can add the `iptables` type here and let `defaults.go` reference it.

**Comment by m1093782566** [comment]
@bowei

I pushed updates now, PTAL.

File: pkg/proxy/apis/kubeproxyconfig/v1alpha1/types.go [hunk 2]
```diff
@@ -148,14 +148,11 @@ type KubeProxyConfiguration struct {
 	ConfigSyncPeriod metav1.Duration `json:"configSyncPeriod"`
 }
 
-// Currently two modes of proxying are available: 'userspace' (older, stable) or 'iptables'
-// (newer, faster). If blank, use the best-available proxy (currently iptables, but may
-// change in future versions).  If the iptables proxy is selected, regardless of how, but
-// the system's kernel or iptables versions are insufficient, this always falls back to the
-// userspace proxy.
+// Currently, four modes of proxying are available total: 'userspace' (older, stable), 'iptables'
```
Comment (thockin) [inline pkg/proxy/apis/kubeproxyconfig/v1alpha1/types.go]: kube-proxy should break into 2 binaries - linux and windows - the flags and docs are so totally disjoint.

File: pkg/proxy/apis/kubeproxyconfig/v1alpha1/types.go [hunk 3]
```diff
@@ -148,14 +148,11 @@ type KubeProxyConfiguration struct {
...
+// (newer, faster), 'ipvs', and 'kernelspace' (Windows only, newer).
+//
+// If blank, use the best-available proxy (currently iptables, but may change in
```
Comment (thockin) [inline pkg/proxy/apis/kubeproxyconfig/v1alpha1/types.go]: This needs to call out the difference between Windows and Linux.  iptables doesn't make sense on Windows.

File: pkg/proxy/apis/kubeproxyconfig/v1alpha1/types.go (hunk 2 above)
Comment (m1093782566) [reply pkg/proxy/apis/kubeproxyconfig/v1alpha1/types.go]: I fixed it now. PTAL.

File: pkg/proxy/apis/kubeproxyconfig/v1alpha1/types.go (hunk 3 above)
Comment (m1093782566) [reply pkg/proxy/apis/kubeproxyconfig/v1alpha1/types.go]: I fixed it now. PTAL.

**Comment by m1093782566** [comment]
@thockin 

All comments are addressed now. Please a look when you have a chance~

**Pull request merged**